src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala

```diff
@@ -421,7 +421,22 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash with Expon
             goto(EnrouteRefueling) using data.asInstanceOf[T]
           } else goto(ConnectingToChargingPoint) using data.asInstanceOf[T]
         } else {
-          handleUseParkingSpot(tick, currentBeamVehicle, id, geo, eventsManager)
+          val maybePersonData = findPersonData(data)
+          val maybeNextActivity =
+            maybePersonData.flatMap(personData => this.asInstanceOf[PersonAgent].nextActivity(personData))
```

Comment:
What gives the certainty that this is a PersonAgent? It seems like a potential for a runtime exception?